cesql/cesql_tck/subscriptions_api_recreations.yaml

```diff
@@ -0,0 +1,176 @@
+    result: true
+    eventOverrides:
+      myext: "customext"
+  - name: Prefix filter on missing string extension
+    expression: "myext LIKE 'custom%'"
+    error: missingAttribute
+
+  - name: Suffix filter (1)
+    expression: "type like '%.error'"
+    result: true
+    eventOverrides:
+      type: "com.github.error"
+  - name: Suffix filter (2)
+    expression: "type like '%.error'"
+    result: false 
+    eventOverrides:
+      type: "com.github.success"
+  - name: Suffix filter on string extension
+    expression: "myext LIKE '%ext'"
+    result: true
+    eventOverrides:
+      myext: "customext"
+  - name: Suffix filter on missing string extension
+    expression: "myext LIKE '%ext'"
+    error: missingAttribute
+
+  - name: Exact filter (1)
+    expression: "id = 'myId'"
+    result: true
+    eventOverrides:
+      id: "myId"
+  - name: Exact filter  (2)
+    expression: "id = 'myId'"
+    result: false
+    eventOverrides:
+      id: "notmyId"
+  - name: Exact filter on string extension
+    expression: "myext = 'customext'"
+    result: true
+    eventOverrides:
+      myext: "customext"
+  - name: Exact filter on missing string extension
+    expression: "myext = 'customext'"
+    error: missingAttribute
+
+  - name: Prefix filter AND Suffix filter (1)
+    expression: "id LIKE 'my%' AND source LIKE '%.ca'"
+    result: true
+    eventOverrides:
+      id: "myId"
+      source: "http://www.some-website.ca"
+  - name: Prefix filter AND Suffix filter (2)
+    expression: "id LIKE 'my%' AND source LIKE '%.ca'"
+    result: false
+    eventOverrides:
+      id: "myId"
+      source: "http://www.some-website.com"
+  - name: Prefix filter AND Suffix filter (3)
+    expression: "myext LIKE 'custom%' AND type LIKE '%.error'"
+    result: true
+    eventOverrides:
+      myext: "customext"
+      type: "com.github.error"
+  - name: Prefix AND Suffix filter (4)
+    expression: "type LIKE 'example.%' AND myext LIKE 'custom%'"
+    error: missingAttribute
+    eventOverrides:
+      type: "example.event.type"
+
+  - name: Prefix OR Suffix filter (1)
+    expression: "id LIKE 'my%' OR source LIKE '%.ca'"
+    result: true
+    eventOverrides:
+      id: "myId"
+      source: "http://www.some-website.ca"
+  - name: Prefix OR Suffix filter (2)
+    expression: "id LIKE 'my%' OR source LIKE '%.ca'"
+    result: true
+    eventOverrides:
+      id: "myId"
+      source: "http://www.some-website.com"
+  - name: Prefix OR Suffix filter (3)
+    expression: "id LIKE 'my%' OR source LIKE '%.ca'"
+    result: true
+    eventOverrides:
+      id: "notmyId"
+      source: "http://www.some-website.ca"
+  - name: Prefix OR Suffix filter (4)
+    expression: "id LIKE 'my%' OR source LIKE '%.ca'"
+    result: false 
+    eventOverrides:
+      id: "notmyId"
+      source: "http://www.some-website.com"
+  - name: Prefix OR Suffix filter (5)
+    expression: "id LIKE 'my%' OR myext LIKE '%ext'"
+    error: missingAttribute
```

Comment:
Just curious, does the spec say that the stuff after the OR must be evaluated if the first part evaluates to 'true'? We should add a test for that, unless I missed it. In particular, if the 2nd part would generate an error if evaluated... what should happen if someone chooses to NOT evaluate it ?

Reply:
I'm not 100% sure, what I could find was:

> Because every operator and function is total, an expression evaluation flow is defined statically and cannot be modified by expected or unexpected errors. Nevertheless CESQL includes the concept of errors: when an expression is evaluated, in case an error arises, the evaluator collects a list of errors, referred in this spec as error list, which is then returned together with the evaluated value of the CESQL expression.

So, I guess the spec does not explicitly specify whether or not there should be an error if the `OR` has an early return. Do you think we should specify this?

Reply:
I sure does seem like the spec should be clear on what `true OR error` results in... either `true` or `error`. Otherwise we'll have an interop issue. I would think most people would expect `true` but I'm not an expert in what query languages normally do for this. @lionelvillard @embano1 any thoughts?

Reply:
@duglin on the go sdk v2, when evaluating `(type = 'example.knative.dev') OR (missingattribute = 'test')` on an event with `type: 'example.knative.dev'` but no `missingattribute` I get: `missing attribute 'missingattribute'`

So, it seems like the current behaviour (at least for go) is that it reports the error, even if the first part evaluated to true

Reply:
Should I open an issue to clarify this behaviour, or do we want to open an issue to change this behaviour and make it behave like you think most people would expect (`true` instead of `error`)?

Reply:
It seems like from https://github.com/cloudevents/spec/blob/main/cesql/spec.md#41-error-handling that we would always return the error regardless of the result though (either fail fast, or continue and find the result and then return result and error)

Reply:
I think we should open an issue to get feedback from the group and then make it clear in the spec... and then align the go-sdk if needed.